### PR TITLE
feat(travis): switch to codecov for code coverage

### PR DESCRIPTION
**Pull request by NBonaparte**

### CODE REVIEW

**Comment by NBonaparte** [comment]
@patrick96 much more informative than coveralls IMO. I'm probably gonna turn off the PR comments.

File: .travis.yml [hunk 1]
```diff
@@ -20,11 +20,13 @@ matrix:
...
-      before_install:
-        - pip install --user cpp-coveralls
       script: ${TRAVIS_BUILD_DIR}/common/travis/tests.sh
       after_success:
-        - coveralls --root .. -E ".*CMakeFiles.*" -e tests -e lib -e build/googletest-src --gcov-options '\-p'
+        - cd ${TRAVIS_BUILD_DIR}
+        - lcov --directory . --capture --output-file coverage.info
+        - lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter out system
+        - lcov --list coverage.info
+        - bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
```
Comment (patrick96) [inline .travis.yml]: The codecov script doesn't actually use any results from `lcov`, it calls gcov itself. Using `lcov` here is purely for aesthetical reasons inside the travis report. I'm fine with that, but I wouldn't really look at the travis log unless something went wrong, so I'm also fine with removing it.

**Comment by patrick96** [comment]
Looks good. Do you think we should have the test code be part of the coverage?

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ignore:
+  - "test/*"
```
Comment (patrick96) [inline .codecov.yml]: The folder is called `tests`

File: README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Polybar
 
 [![Build Status](https://travis-ci.org/jaagr/polybar.svg?branch=master)](https://travis-ci.org/jaagr/polybar)
-[![Coverage Status](https://coveralls.io/repos/github/jaagr/polybar/badge.svg?branch=master)](https://coveralls.io/github/jaagr/polybar?branch=master)
+[![Coverage Status](https://codecov.io/gh/jaagr/polybar/branch/master/graph/badge.svg)](https://codecov.io/gh/jaagr/polybar/branch/master/graph/badge.svg)
```
Comment (patrick96) [inline README.md]: Please make the badge link directly to the codecov page. After that, I think this is ready to merge

**Pull request merged**